src/Form/Type/ImmutableTabsType.php

```diff
@@ -24,16 +25,23 @@
  *
  * @author Benoit de Jacobet <benoit.de-jacobet@ekino.com>
  */
-class ImmutableTabsType extends ImmutableArrayType
+class ImmutableTabsType extends AbstractType
 {
-    /**
-     * {@inheritdoc}
-     */
+    private $immutableArrayType;
+
+    public function __construct(ImmutableArrayType $immutableArrayType)
+    {
+        $this->immutableArrayType = $immutableArrayType;
+    }
+
     public function configureOptions(OptionsResolver $resolver)
     {
-        parent::configureOptions($resolver);
+        $resolver->setAllowedTypes('tabs', 'string[]')->setRequired('tabs');
```

Comment:
Missing call to "parent" I think

Reply:
Parent call on AbstractType is useless, this class will never define anything.

Reply:
@fancyweb no longer needed as we said ?

Reply:
I'm not talking about AbstractType, but about `$this->immutableArrayType`

Reply:
$this->immutableArrayType should not exist.

Reply:
well, before the class was extending the one of Sonata and calls the parent here, so I guess something is missing now

Reply:
but it's ok, Symfony will get the OptionsResolver of the "parent" before passing it to the child, and Sonata does not do anormal things, so it's ok like this